### PR TITLE
"Add Variation" Button

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Address issue #3205 with two changes:

1 - Adapting the Variations `ProductDetailCardBuilder` method to now present the `Add Variation` text instead of the previous `This product has no variation yet` description.

2 - Adding to `VariationListFragment`a new kind of `ActionableEmptyView` that presents the `Add Variation` view and action instead of suggesting the user go to the web.

⚠️ The `Add Variation` button behavior will be implemented on upcoming PRs

How to Test
==========

New behavior (only available when on debug mode)
----------
| Click on Products section and select a Variable Product | Click on `Add Variations` and make sure no view is called instead of the old web redirection suggestion |
| ------------- | ------------- |
| Screenshot_1614251390 | Screenshot_1614247056 |
Old behavior
----------
| Click on Products section and select a Variable Product | Click on `Variations` | Make sure presented View informs the user about adding variations only available on the web |
| ------------- | ------------- | ------------- |
| Screenshot_1614251390 | Screenshot_1614251359 | Screenshot_1614251362 |

Update release notes:

- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by nbradbury** [comment]
@ThomazFB This is really a design issue, but it seems awkward to tap "Add variations" in the product detail screen and be taken to another screen where you have to tap again to add a variation. Also, the button on the second screen says "Add attribute" - shouldn't that be "Add variation?"

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt [hunk 1]
```diff
@@ -232,4 +241,13 @@ class VariationListFragment : BaseFragment(R.layout.fragment_variation_list),
 
         adapter.setVariationList(variations)
     }
+
+    private fun ActionableEmptyView.handleEmptyViewVisibility(shouldBeVisible: Boolean, showButton: Boolean) {
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt]: Would it make more sense to move this into `ActionableEmptyView`? Also, I think we'd want to check the visibility before running the animation, eg:

```
if (shouldBeVisible && !isVisible) {
     WooAnimUtils.fadeIn(this)
     showButton(showButton)
} else if (!shouldBeVisible && isVisible) {
     WooAnimUtils.fadeOut(this)
 }
```

Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt]: Agreed! Much better this way, I've submitted the changes right here: https://github.com/woocommerce/woocommerce-android/pull/3620/commits/9bbdc18da6eed3bfc9bf11c363aa247397be964c

**Comment by ThomazFB** [comment]
> This is really a design issue, but it seems awkward to tap "Add variations" in the product detail screen and be taken to another screen where you have to tap again to add a variation. 

Yeah, I found that to be a little bit odd too, it does make sense to prompt the `Add your first variation` view when we delete the last Variation available to a given product on the list, but selecting the `Add Attribute` from the Product Details view should direct the user immediately to the Variation creation flow, that was a nice catch @nbradbury! 

@Garance91540 @adamzelinski what are your thoughts about this point Nick suggested?

CC: @Ecarrion 

> Also, the button on the second screen says "Add attribute" - shouldn't that be "Add variation?"

Oh, sorry about that 🤦‍♂️, it should be "Add variation", I've fixed that here: https://github.com/woocommerce/woocommerce-android/pull/3620/commits/ea58518d3e0ab2961661c6cde37945fe7a2a72c6

**Comment by Garance91540** [comment]
Hi there, in my initial designs, when selecting "Add variations" on the product page, users where directly taken to the ["add attribute" screen](https://d.pr/i/k1nHQY) as you suggested. But because of the way the navigation is set up on iOS, we had to add back that empty state "[add variation](https://d.pr/i/CC72Tf)" screen. So if it's possible to skip it in Android, that's awesome! 

So we should only display this screen if a user has deleted all variations, and you are right @nbradbury the CTA should be "Add Variation" and not "Add Attribute"

**Comment by Ecarrion** [comment]
> So we should only display this screen if a user has deleted all variations, and you are right @nbradbury the CTA should be "Add Variation" and not "Add Attribute"

FYI: It should also be shown if the user has deleted all attributes, as core won't delete the variations for you!

**Comment by ThomazFB** [comment]
Hey @nbradbury, the PR is ready for another round, but since now we changed the behavior of the `Add Variation` button at Product Details View as we discussed here, and currently we don't support deleting variations from the app, it's worth noting that the new `Add your first variation` view prompt at the Variation List View is no longer reachable and it will be back only after we fully support managing the Product Variation list, nevertheless I'm leaving here the code for that view since it's behind a feature flag and it's correctly functional for upcoming PRs.

I've changed the PR description to correctly represent the currently available flow for testing, which is much smaller now.

Also, thanks @Garance91540 and @Ecarrion for the awesome feedbacks and tips!

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**